### DISCUSSION TITLE
using CanvasGradient or CanvasPattern with style expressions

### DISCUSSION THREAD

**Discussion by TheWeirdDev** [openlayers/openlayers#15731] [Q&A] [answered]
Hi,
I'm trying to port our old codebase to the latest version of OL. I'm interested in using the new style expression syntax but I've hit a roadblock. The old code uses gradients and patterns for some styles and I can't figure out how to port it to style expressions.

Is it possible to use CanvasGradient or CanvasPattern with style expressions?

**Comment by ahocevar** [comment, marked as answer]
See https://github.com/openlayers/openlayers/pull/15318 - you can use `fill-pattern-src`, `fill-pattern-size` and `fill-pattern-offset`.

**Comment by TheWeirdDev** [comment]
Awesome. Thanks!
Is there any place I could find all supported style properties? I didn't find these in the docs.

**Comment by ahocevar** [comment]
You can find them here: https://openlayers.org/en/latest/apidoc/module-ol_style_flat.html. Or through the navigation tree: `ol/style/flat`.
